L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -1,21 +1,106 @@
 #include "L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorSuperTriggerCellImpl.h"
+#include "DataFormats/ForwardDetId/interface/HGCalTriggerDetId.h"
 
 #include <unordered_map>
 
 HGCalConcentratorSuperTriggerCellImpl::
-HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf){}
+HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf)
+  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize"))
+{
 
+    if ( stcSize_.size() != kNLayers_ ){
+        throw cms::Exception("HGCTriggerParameterError")
+            << "Inconsistent size of super trigger cell size vector" << stcSize_.size() ;
+    }
+    for(auto stc : stcSize_) {
+        if ( stc!=kSTCsizeFine_ && stc!=kSTCsizeCoarse_ ){
+            throw cms::Exception("HGCTriggerParameterError")
+              << "Super Trigger Cell should be of size 4 or 16" ;
+        }
+    }
+    
+}
+
+std::map<int,int> 
+HGCalConcentratorSuperTriggerCellImpl::kSplit_ = {
+  {4, 0x3a},
```

Comment:
use `kSTCsizeFine_` and `kSTCsizeCoarse_` for keys
should values also be named constants?